src/partition/repart.c

```diff
@@ -1400,6 +1401,52 @@ static int config_parse_make_dirs(
         }
 }
 
+static int config_parse_make_subvolumes(
```

Comment:
this function is 99% identical to config_parse_make_directories(), no?

the only difference I can see is in the error string, but you can easily fix that by simply using the `lvalue` parameter in the message string.